plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -471,18 +472,18 @@ public static void verifySupportedColumnMapping(ColumnMappingMode mappingMode)
         }
     }
 
-    public static List<DeltaLakeColumnMetadata> getColumnMetadata(String json, TypeManager typeManager, ColumnMappingMode mappingMode)
+    public static List<DeltaLakeColumnMetadata> getColumnMetadata(String json, TypeManager typeManager, ColumnMappingMode mappingMode, List<String> partitionColumns)
     {
         try {
             ImmutableList.Builder<DeltaLakeColumnMetadata> columns = ImmutableList.builder();
             Iterator<JsonNode> nodes = OBJECT_MAPPER.readTree(json).get("fields").elements();
             while (nodes.hasNext()) {
                 try {
-                    columns.add(mapColumn(typeManager, nodes.next(), mappingMode));
+                    columns.add(mapColumn(typeManager, nodes.next(), mappingMode, partitionColumns));
                 }
                 catch (UnsupportedTypeException e) {
                     // Write operations are denied by unsupported 'variantType' writer feature
```

Comment:
The comment is not corresponding anymore only to variant - the catch statement deals now also with unsupported type widenings.